hw/dv/tools/vcs/common_cov_excl.cfg

```diff
@@ -11,6 +11,8 @@
 -node tb.dut *tl_o.d_opcode[1]
 -node tb.dut *tl_o.d_opcode[2]
 -node tb.dut *tl_o.d_sink
+// [UNR] due to the ECC logics
+-node tb.dut *tl_o.d_user.rsp_intg[6]
```

Comment:
@tjaychen Due to ECC we use, this bit is a fixed value,  so it's effectively 5 bits intg value. I think it's fine. LMK if you have any concern. Thanks

Reply:
i'm a bit confused, aren't we using a 7-bit ECC? why do we say this is always a fixed value? 
Or do you mean based on the way the response is padded out, this always ends up being 0?

Reply:
@tjaychen 
sorry, it's 7 bits. The bit[6] is also a fixed value.

The rsp data used for ECC is only 4 bits wide.
```
  function automatic tl_d2h_rsp_intg_t extract_d2h_rsp_intg(tl_d2h_t tl);
    tl_d2h_rsp_intg_t payload;
    logic unused_tlul;
    unused_tlul = ^tl;
    payload.opcode = tl.d_opcode; // only bit0 can be toggle, as we only return accessAck(0) or accessAckData(1)
    payload.size   = tl.d_size; // 2 bits wide
    //payload.source = tl.d_source;
    payload.error  = tl.d_error;
    return payload;
  endfunction // extract_d2h_rsp_intg
```

See my comment below.
```
module prim_secded_inv_64_57_enc (
  input        [56:0] data_i,
  output logic [63:0] data_o
);

  always_comb begin : p_encode
    data_o = 64'(data_i);
    data_o[57] = 1'b0 ^ ^(data_o & 64'h0103FFF800007FFF);
    data_o[58] = 1'b1 ^ ^(data_o & 64'h017C1FF801FF801F);
    data_o[59] = 1'b0 ^ ^(data_o & 64'h01BDE1F87E0781E1);
    data_o[60] = 1'b1 ^ ^(data_o & 64'h01DEEE3B8E388E22);
    data_o[61] = 1'b0 ^ ^(data_o & 64'h01EF76CDB2C93244);
    data_o[62] = 1'b1 ^ ^(data_o & 64'h01F7BB56D5525488);
    data_o[63] = 1'b0 ^ ^(data_o & 64'h01FBDDA769A46910); // the lower 4 bits are 0, while `data_o` only have values on the lower 4 bits, so this data_o[63] is always 0.
  end

```

Reply:
I wonder whether this means we should explicitly drop a bit from the signal? It would probably be clearer to do that than to pass around a known zero. @tjaychen: Would that be fiddly or is it a simple matter of a `s/7/6/g`? :-

Reply:
my main worry is that if we ever change encoding, it might be another bit that becomes fixed and not 6 (or if we add more bits into the response). Because the mapping of which bit to pick into the calculation is effectively random right? Or is there some mathematical proof here that suggests this will always be 0?

Reply:
IMO, since rsp is only 4 bits wide, we probably don't need so many bits as its integrity. I assume synthesis will optimize this bit[6] away. If so, I guess we can just exclude it in DV. This 7 bit integrity scheme has also been used for others like cmd and data. We may not need to drop it to 6 bits.

Or, we can below lower 4 bits to a non-zero value.
```data_o[63] = 1'b0 ^ ^(data_o & 64'h01FBDDA769A46910);```

> Because the mapping of which bit to pick into the calculation is effectively random right? Or is there some mathematical proof here that suggests this will always be 0?

UNR can tell us a mathematical proof result.

Reply:
sorry what i meant this is effectively random, so i think if we were ever to regenerate, the results might be different. 
What i meant by mathematic proof is whether we can claim that if input is only 4-bits padded with 0, certain bits will always be a fixed value (i don't think so..but not sure).

Reply:
As discussed offline with Tim, if we regenerate with another ECC scheme, it could be some other bits become unreachable or maybe no unreachable at all.
With current ECC logics, bit 6 is UNR and we can exclude it as it should be optimized away by synthesis tool.

Reply:
That's a good point. I just assumed it was something more fundamental because the fixed bit happened to be the top one. As you say, let's go with this plan.